tests/cnf/core/network/metallb/tests/bgp-unnumbered.go

```diff
@@ -0,0 +1,282 @@
+
+			By("Creating a new instance of MetalLB Speakers on workers")
+			err := metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create metalLb daemonset")
+
+			By("Collecting interface information to enable a node ethernet interface")
+			srIovInterfacesUnderTest, err = NetConfig.GetSriovInterfaces(1)
+			Expect(err).ToNot(HaveOccurred(), "Failed to retrieve SR-IOV interfaces for testing")
+
+			By(fmt.Sprintf(
+				"nmstate policy configuring ethernet interface %s on %s with an IP address",
+				srIovInterfacesUnderTest[0], workerNodeList[0].Definition.Name))
+			enableEthernetInterfaceWithIP(nodeNetConfigPolicyName, workerNodeList[0].Definition.Name,
+				srIovInterfacesUnderTest[0])
+
+			By("Collecting the frrk8sPod Pod list for worker nodes in list cnfWorkerNodeList")
+			frrk8sPods = verifyAndCreateFRRk8sPodList()
+
+			By("Creating BFD profile.")
+			createBFDProfileAndVerifyIfItsReady(frrk8sPods)
+
+			By("Creating host-device NAD for external FRR pod")
+			_, err = define.HostDeviceNad(APIClient, frrconfig.ExternalMacVlanNADName, srIovInterfacesUnderTest[0],
+				tsparams.TestNamespaceName, nad.IPAMStatic())
+			Expect(err).ToNot(HaveOccurred(), "Failed to create a network-attachment-definition")
+
+			By("Create a single IPAddressPool")
+			ipAddressPool := createIPAddressPool("ipaddresspool", ipAddressPoolRange)
+
+			By("Creating a MetalLB service")
+			setupMetalLbService(tsparams.MetallbServiceName, netparam.IPV4Family, tsparams.LabelValue1,
+				ipAddressPool, "Cluster")
+
+			By(fmt.Sprintf(
+				"Creating a BGPAdvertisement with the nodeSelector to %s", workerNodeList[0].Definition.Name))
+			worker0NodeLabel := []metav1.LabelSelector{
+				{MatchLabels: map[string]string{corev1.LabelHostname: workerNodeList[0].Definition.Name}},
+			}
+
+			setupBgpAdvertisement("bgpadvertisment", tsparams.CustomCommunity, ipAddressPool.Definition.Name,
+				200, []string{tsparams.BgpPeerName1}, worker0NodeLabel)
+		})
+
+		AfterAll(func() {
+			By("Remove custom MetalLB test label from nodes")
+			removeNodeLabel(workerNodeList, metalLbTestsLabel)
+
+			By(fmt.Sprintf("Disabling ethernet interface %s on worker node %s",
+				srIovInterfacesUnderTest[0], workerNodeList[0].Definition.Name))
+			ethIntWorker0Policy := nmstate.NewPolicyBuilder(APIClient, nodeNetConfigPolicyName, NetConfig.WorkerLabelMap).
+				WithAbsentInterface(srIovInterfacesUnderTest[0])
+			err := netnmstate.UpdatePolicyAndWaitUntilItsAvailable(netparam.DefaultTimeout, ethIntWorker0Policy)
+			Expect(err).ToNot(HaveOccurred(), "Failed to update NMState network policy")
+
+			By("Removing NMState policies")
+			err = nmstate.CleanAllNMStatePolicies(APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove all NMState policies")
+
+			By("Clean MetalLB operator and test namespaces")
+			resetOperatorAndTestNS()
+		})
+
+		AfterEach(func() {
```

Comment:
Some times the aftereach might fail not executed, it worth having a before each saying "is my cluster clean".
Even though I would say explore the possibility if all the specs can ran in parallel

Reply:
All are test cases are build with the same format. I will bring this up in my next team meeting with Evgeny and Anvesh.